### PR TITLE
Update dependency versions

### PR DESCRIPTION
**Pull request by tgauth**
# PR Summary
- bump ZLib version to `1.3.1`
- bump LibFido2 version to `1.15.0`
- bump LibreSSL version to `3.9.2.0`
- add `arc4random_win.h` and use OpenBSD compat's `arc4random` since LibreSSL has stopped exporting symbols per https://github.com/libressl/portable/issues/1087

### CODE REVIEW

File: contrib/win32/openssh/config.h.vs [hunk 1]
```diff
@@ -207,13 +207,13 @@
 /* #undef HAVE_ADDR_V6_IN_UTMPX */
 
 /* Define to 1 if you have the `arc4random' function. */
-#define HAVE_ARC4RANDOM 1
```
Comment (vthiebaut10) [inline contrib/win32/openssh/config.h.vs:210]: Why are these commented out?

File: contrib/win32/win32compat/misc.c [hunk 2]
```diff
@@ -191,7 +191,7 @@ nanosleep(const struct timespec *req, struct timespec *rem)
  * Copyright (c) 2009, 2010 NoMachine
  * All rights reserved
  */
-static int
```
Comment (vthiebaut10) [inline contrib/win32/win32compat/misc.c:194]: why are these not static anymore?

File: contrib/win32/openssh/config.h.vs (hunk 1 above)
Comment (tgauth) [reply contrib/win32/openssh/config.h.vs:210]: The symbols are no longer exported by LibreSSL, so we need to compile them at runtime in openbsd-compat.

Comment (vthiebaut10) [reply contrib/win32/openssh/config.h.vs:210]: should we just remove them then?

File: contrib/win32/win32compat/misc.c (hunk 2 above)
Comment (tgauth) [reply contrib/win32/win32compat/misc.c:194]: Also related to the symbols no longer being exported by LibreSSL. Updating to match internal declarations:
https://github.com/PowerShell/openssh-portable/blob/latestw_all/contrib/win32/win32compat/inc/strings.h
https://github.com/PowerShell/openssh-portable/blob/latestw_all/contrib/win32/win32compat/inc/sys/time.h

**Review by vthiebaut10 — APPROVED** [review]

**Pull request merged**